### PR TITLE
feature/tags

### PR DESCRIPTION
**Pull request by k-nut**

### CODE REVIEW

File: forms/migrations/0007_auto_20180418_1504.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('forms', '0006_investigation_data_privacy_url'),
+    ]
+
+    operations = [
```
Comment (k-nut) [inline forms/migrations/0007_auto_20180418_1504.py]: @sjockers @lisaq where is this coming from? 🤔

File: forms/migrations/0008_merge_20180419_1606.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Generated by Django 2.0.1 on 2018-04-19 16:06
```
Comment (k-nut) [inline forms/migrations/0008_merge_20180419_1606.py]: I think if we drop those weird e-mail migrations we might also not need this file.

File: forms/models.py [hunk 3]
```diff
@@ -311,6 +321,15 @@ def json_email(self):
...
         return self.form_instance.form.investigation.slug == investigation_slug
 
+    @property
+    def taglist(self):
+        investigation = self.form_instance.form.investigation
+        return list(Tag.objects.filter(investigation=investigation).all())
+
+    @property
+    def selected_tags(self):
+        return list(self.tags.all())
```
Comment (k-nut) [inline forms/models.py]: I think that you could drop this function and just use `for tag in form_response.tags` in the template, couldn't you?

File: forms/migrations/0007_auto_20180418_1504.py (hunk 1 above)
Comment (lisaq) [reply forms/migrations/0007_auto_20180418_1504.py]: \(o_o)/ manage.py did it

File: forms/models.py (hunk 3 above)
Comment (lisaq) [reply forms/models.py]: 🤔 hrm perhaps

File: forms/tests/test_tags.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+                                   TagFactory)
+
+
+class TagTest(TestCase):
+
+    def test_tag_in_response(self):
+        tag = TagFactory.create()
+        form_response = FormResponseFactory.create()
+        form_response.tags.set([tag])
+        self.assertEqual(form_response.selected_tags, [tag])
```
Comment (lisaq) [inline forms/tests/test_tags.py]: ...but then...

**Comment by k-nut** [comment]
@lisaq is there any way to unselect a tag once it was selected?

**Comment by lisaq** [comment]
@k-nut sure, the same way you select them - in the response detail view

**Pull request merged**